### PR TITLE
use custom toolchains for travis-ci

### PR DESCRIPTION
**Pull request by pazos**
on pocketbook we download the sdk and then [export the path](https://github.com/koreader/virdevenv/blob/master/docker/ubuntu/kopb/Dockerfile#L8).

for kindles, kobos and cervantes we build koxtoolchain but actually don't use it. (I see in the logs calls to arm-linux-gnueabi / arm-linux-gnueabihf, which implies that our custom toolchains were not found on PATH.

For example, for kobos the toolchain check in koreader-base is [here](https://github.com/koreader/koreader-base/blob/master/Makefile.defs#L61-L66)

### CODE REVIEW

**Comment by Frenzie** [comment]
> I see in the logs calls to arm-linux-gnueabi / arm-linux-gnueabihf, which implies that our custom toolchains were not found on PATH.

That's true for the cervantes Travis log, but neither for Kobo nor Kindle.

```
cd thirdparty/luajit/build/arm-kobo-linux-gnueabihf && \
		cmake -DCC="gcc" \
		-DXCOMPILE:BOOL=on \
		-DBASE_CFLAGS="-O2 -ffast-math -pipe -fomit-frame-pointer" -DHOST_CFLAGS="-march=native -O2 -ffast-math -pipe -fomit-frame-pointer " \
		 \
		 \
		-DTARGET_SONAME=libluajit.so \
		-DTARGET_CFLAGS="-O2 -ffast-math -pipe -fomit-frame-pointer  -march=armv7-a -mtune=cortex-a8 -mfpu=neon -mthumb -mfloat-abi=hard -fno-finite-math-only -fno-stack-protector -U_FORTIFY_SOURCE -fPIC -DKOBO_PLATFORM" \
		-DCROSS="/usr/bin/ccache arm-kobo-linux-gnueabihf-" \
```

```
-- The C compiler identification is GNU 7.2.1
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /home/ko/x-tools/arm-kobo-linux-gnueabihf/bin/arm-kobo-linux-gnueabihf-gcc
-- Check for working C compiler: /home/ko/x-tools/arm-kobo-linux-gnueabihf/bin/arm-kobo-linux-gnueabihf-gcc -- works
```

**Comment by Frenzie** [comment]
@pazos 
Alright, I'm at a proper computer now. It's in the base image:

https://github.com/koreader/virdevenv/blob/68cd0004434754078ffce801c6386fdde2dbfc21/docker/ubuntu/baseimage/.bashrc#L13-L14

**Comment by pazos** [comment]
@Frenzie thanks for your time and help!

Is it ok to make changes to the kocervantes Dockerfile or do I need to modify the base image?

**Comment by Frenzie** [comment]
Pinging @houqp who set it up for some thoughts, as well as @NiLuJe who knows all about the TC even though that's only more tangentially related. ;-)

Since it is in there I'd say we should use the base image, but the main reason I didn't do it myself yet last night after tracking it down, is because architecturally speaking I haven't made up my mind yet. My current order of preference after sleeping on it for a night looks like this:

1. Use globbing for forward compatibility with additional custom TCs. However, this should be used with a comment because it won't be as self-explanatory or easy to find as this was. Something like this:
```bash
for i in ${XTOOLS}/*/bin; do
  PATH=$i:$PATH;
done
```
2. Just add it to the line in base image.
3. Switch to doing it in individual images.

**Comment by NiLuJe** [comment]
AFAIK, the x-tools directory is pretty much only used by ct-ng, so that, at least would NOT be something to hold against 1. ;).

TL;DR: 1 & 2 both make sense.

3 possibly makes the intent more obvious, and less cumbersome in the extremely unlikely something changes PATH/triple-wise.

**Comment by Frenzie** [comment]
I'm not sure if 3 is actually less cumbersome. Sure, you have to update one less Docker image, but instead you have to update twice as many files by hand. I think I'll go for 1. :)

**Comment by Frenzie** [comment]
Superseded by https://github.com/koreader/virdevenv/pull/19
